crates/m3/src/gadgets/hash/groestl.rs

```diff
@@ -326,9 +327,7 @@ struct SBox<const V: usize> {
 
 impl<const V: usize> SBox<V> {
 	pub fn new(table: &mut TableBuilder, input: Expr<B8, V>) -> Self {
-		let b8_basis: [_; 8] = array::from_fn(|i| {
-			<B8 as ExtensionField<B1>>::basis(i).expect("i in range 0..8; extension degree is 8")
-		});
+		let b8_basis: [_; 8] = array::from_fn(ext_basis::<B8, B1>);
```

Comment:
If this is used also used in other places we could declare something like
```rust
pub(crate) const B8_B1_BASIS: std::sync::LazyLock<[B8; 8]> = std::sync::LazyLock::new(|| {
   array::from_fn(ext_basis::<B8, B1>);
});
```

Reply:
This call should be very fast and is not in a very hot loop, so I'm in favor of leaving it rather than using a lazy static.